File: check_api_payment.py

import requests
import json

try:
    response = requests.get('http://localhost:5000/api/v1/students/696ea84d6aa1df5c5e08e6d4/dashboard')
    data = response.json()
    
    if data.get('success'):
        payment_data = data['data']['payment']
        print(json.dumps(payment_data, indent=2))
        
        # Verify history length
        print(f"History Length: {len(payment_data.get('history', []))}")
    else:
        print("API returned success: false")
        print(data)

except Exception as e:
    print(f"Error: {e}")

File: check_api_payment.js

const http = require('http');

http.get('http://localhost:5000/api/v1/students/696ea84d6aa1df5c5e08e6d4/dashboard', (res) => {
    let data = '';
    res.on('data', (chunk) => {
        data += chunk;
    });
    res.on('end', () => {
        try {
            const json = JSON.parse(data);
            console.log(JSON.stringify(json.data.payment, null, 2));
        } catch (e) {
            console.error(e);
        }
    });
}).on('error', (err) => {
    console.error('Error: ' + err.message);
});
